storage/tokudb/tokudb_debug.h

```diff
@@ -132,6 +132,61 @@ static void tokudb_backtrace(void);
     DBUG_VOID_RETURN; \
 }
 
+#define TOKUDB_SHARE_TRACE(_fmt, ...) \
+    fprintf(stderr, "%u %p %s:%u TOUDB_SHARE::%s " _fmt "\n", \
+            tokudb::thread::my_tid(), this, __FILE__, __LINE__, \
+            __FUNCTION__, ##__VA_ARGS__);
+
+#define TOKUDB_SHARE_TRACE_FOR_FLAGS(_flags, _fmt, ...) { \
+    if (TOKUDB_UNLIKELY(TOKUDB_DEBUG_FLAGS(_flags))) { \
+        TOKUDB_SHARE_TRACE(_fmt, ##__VA_ARGS__); \
+    } \
+}
+
+#define TOKUDB_SHARE_DBUG_ENTER(_fmt, ...) { \
+    if (TOKUDB_UNLIKELY((tokudb::sysvars::debug & TOKUDB_DEBUG_ENTER) || \
+        (tokudb::sysvars::debug & TOKUDB_DEBUG_SHARE))) { \
+        TOKUDB_SHARE_TRACE(_fmt, ##__VA_ARGS__); \
+    } \
+} \
+    DBUG_ENTER(__FUNCTION__);
+
+#define TOKUDB_SHARE_DBUG_RETURN(r) { \
+    int rr = (r); \
+    if (TOKUDB_UNLIKELY((tokudb::sysvars::debug & TOKUDB_DEBUG_RETURN) || \
+        (tokudb::sysvars::debug & TOKUDB_DEBUG_SHARE) || \
+        (rr != 0 && (tokudb::sysvars::debug & TOKUDB_DEBUG_ERROR)))) { \
+        TOKUDB_SHARE_TRACE("return %d", rr); \
+    } \
+    DBUG_RETURN(rr); \
+}
+
+#define TOKUDB_SHARE_DBUG_RETURN_DOUBLE(r) { \
```

Comment:
Yeah, I know, but it was the paradigm before. I have in mind something similar to what you suggest to replace it all. The TOKU debug could stand another cleaning but it is all out of scope really for right now. I had to add some of this to try and figure out this locking/eviction issue and thought it would be a wast to throw it away since it is mostly here now..